File: source/kCura.Relativity.Export/Types/ViewFieldInfo.vb
Imports RelViewFieldInfo = Relativity.ViewFieldInfo

Namespace kCura.Relativity.Export.Types

	<Serializable()> Public Class ViewFieldInfo
		Inherits RelViewFieldInfo

		Implements IComparable

		Public Sub New(ByVal row As System.Data.DataRow)
			MyBase.New(row)
		End Sub
		Public Sub New(ByVal vfi As RelViewFieldInfo)
			MyBase.New(vfi)
		End Sub

		Public Overrides Function ToString() As String
			Return Me.DisplayName
		End Function

		Public Function CompareTo(ByVal obj As Object) As Integer Implements System.IComparable.CompareTo
			Return String.Compare(Me.DisplayName, obj.ToString)
		End Function

		Public Shadows Function Equals(ByVal other As ViewFieldInfo) As Boolean
			If Me.AvfId = other.AvfId AndAlso Me.AvfColumnName = other.AvfColumnName Then Return True
			Return False
		End Function

		Public Overrides Function GetHashCode() As Integer
			Return 45 * Me.AvfId
		End Function
	End Class

End Namespace

File: source/kCura.Relativity.Export.ConvertedToC#/Types/ViewFieldInfo.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Data;
using System.Linq;
using RelViewFieldInfo = Relativity.ViewFieldInfo;

namespace kCura.Relativity.Export.Types
{

	[Serializable()]

	public class ViewFieldInfo : RelViewFieldInfo, IComparable
	{

		public ViewFieldInfo(System.Data.DataRow row) : base(row)
		{
		}
		public ViewFieldInfo(RelViewFieldInfo vfi) : base(vfi)
		{
		}

		public override string ToString()
		{
			return this.DisplayName;
		}

		public int CompareTo(object obj)
		{
			return string.Compare(this.DisplayName, obj.ToString());
		}

		public new bool Equals(ViewFieldInfo other)
		{
			if (this.AvfId == other.AvfId && this.AvfColumnName == other.AvfColumnName)
				return true;
			return false;
		}

		public override int GetHashCode()
		{
			return 45 * this.AvfId;
		}
	}

}
